### PR TITLE
Remove caching store

### PR DESCRIPTION
**Pull request by suhasagg**
- [ ] I added unit tests for any code that added
- [ ] I updated the CHANGELOG.md 
- [ ] All IP is original and not copied from another source
- [ ] I assign all copyright to Loom Network for the code in the pull request

This PR removes caching store support from loomchain as it has been superseded by the versionedCachingStore.

### CODE REVIEW

File: cmd/loom/loom.go [hunk 1]
```diff
@@ -617,14 +617,7 @@ func loadAppStore(cfg *config.Config, logger *loom.Logger, targetVersion int64)
...
 
-	if cfg.CachingStoreConfig.CachingEnabled &&
-		((cfg.AppStore.Version == 1) || cfg.CachingStoreConfig.DebugForceEnable) {
-		appStore, err = store.NewCachingStore(appStore, cfg.CachingStoreConfig)
-		if err != nil {
-			return nil, err
-		}
-		logger.Info("CachingStore enabled")
-	} else if cfg.CachingStoreConfig.CachingEnabled && cfg.AppStore.Version == 3 {
+	if cfg.CachingStoreConfig.CachingEnabled && cfg.AppStore.Version == 3 {
```
Comment (pathornteng) [inline cmd/loom/loom.go]: AppStore v1 is now compatible with `VersionedCachingStore` too.

Comment (suhasagg) [reply cmd/loom/loom.go]: AppStore v1 Support is added.

File: store/versioned_cachingstore.go [hunk 2]
```diff
@@ -9,14 +9,138 @@ import (
...
+		}, []string{"cache_operation"})
+
+}
+
+type CachingStoreConfig struct {
+	CachingEnabled bool
+	// CachingEnabled may be ignored in some configurations, this will force enable the caching
+	// store in those cases.
+	// WARNING: This should only used for debugging.
+	DebugForceEnable bool
```
Comment (pathornteng) [inline store/versioned_cachingstore.go]: This configuration is not used anymore so we can remove it.

Comment (suhasagg) [reply store/versioned_cachingstore.go]: Done.

**Pull request merged**